udata_hydra/analysis/csv.py

```diff
@@ -43,18 +50,43 @@
     "float": Float,
     "int": BigInteger,
     "bool": Boolean,
+    "json": JSON,
+    "date": Date,
+    "datetime": DateTime,
 }
 
 PYTHON_TYPE_TO_PY = {
     "string": str,
     "float": float,
     "int": int,
     "bool": bool,
+    "json": helpers.to_json,
+    "date": helpers.to_date,
+    "datetime": helpers.to_datetime,
 }
 
 RESERVED_COLS = ("__id", "tableoid", "xmin", "cmin", "xmax", "cmax", "ctid")
 
 
+async def notify_udata(ca_id):
+    """Notify udata of the result of a parsing"""
+    analysis = await get_csv_analysis(ca_id)
+    resource_id = analysis["resource_id"]
+    db = await context.pool()
+    record = await db.fetchrow("SELECT dataset_id FROM catalog WHERE resource_id = $1", resource_id)
+    if record:
+        payload = {
+            "resource_id": resource_id,
+            "dataset_id": record["dataset_id"],
+            "document": {
+                "analysis:parsing:table": analysis["parsing_table"],
+                "analysis:parsing:error": analysis["parsing_error"],
```

Comment:
I'm adding this here because it occurred for me, but we can remove the pruning of `None` values in `send` now :)